app/src/editor/canvas/services.js

```diff
@@ -151,48 +147,38 @@ export async function loadParentCanvas(canvas) {
     return loadCanvas({ id: settings.parentCanvasId })
 }
 
-export async function unlinkCanvas(canvas) {
-    if (!(canvas && canvas.settings && canvas.settings.childCanvasId)) {
-        return canvas // do nothing if no child canvas set
-    }
-
-    return saveNow({
-        ...canvas,
-        settings: {
-            ...canvas.settings,
-            childCanvasId: undefined,
-        },
-    })
-}
-
 /**
  * Unlinks parent from child.
  */
 
-export async function unlinkParentCanvas(canvas) {
+export async function unlinkAndLoadParentCanvas(canvas) {
     const { settings = {} } = canvas
     const parent = await loadCanvas({ id: settings.parentCanvasId })
-    return unlinkCanvas(parent)
+    unlink(parent.id)
+    return parent
 }
 
 /**
  * Loads child canvas if one is active, otherwise loads passed-in canvas.
  */
 
 export async function loadRelevantCanvas({ id }) {
-    const canvas = await loadCanvas({ id })
-    if (canvas.settings.childCanvasId != null) {
-        const childCanvas = await loadCanvas({ id: canvas.settings.childCanvasId }).catch((error) => {
-            analytics.reportWarning(error, {
-                message: 'error loading child canvas',
-                parentCanvas: canvas.id,
-                childCanvas: canvas.settings.childCanvasId,
-            })
-            return unlinkCanvas(canvas)
-        })
-        return childCanvas
+    const childCanvasId = getLink(id)
```

Comment:
Can you add more explanation here why is linking needed and what is done here?

Reply:
Done in 856e9f4